test_image.sh

```diff
@@ -77,15 +79,31 @@ function prepare_mount {
 
 	cp -r templates/test/* "${TEST_DIR}"
 
-	if [ -n "${LIFERAY_DOCKER_TEST_HOTFIX_URL}" ]
+	mkdir -p "${TEST_DIR}/patching"
+
+	if [ -n "${LIFERAY_DOCKER_TEST_PATCHING_TOOL_URL}" ]
 	then
-		mkdir -p "${TEST_DIR}/patching"
+		local patcing_tool_file_name=${LIFERAY_DOCKER_TEST_PATCHING_TOOL_URL##*/}
+
+		download "downloads/patching-tool/${patcing_tool_file_name}" "${LIFERAY_DOCKER_TEST_PATCHING_TOOL_URL}"
+	else
+		local patcing_tool_file_name=$(basename $(ls -Art downloads/patching-tool/*.zip | tail -n 1))
+	fi
 
+	if [ -n "${LIFERAY_DOCKER_TEST_HOTFIX_URL}" ]
+	then
 		local hotfix_file_name=${LIFERAY_DOCKER_TEST_HOTFIX_URL##*/}
 
 		download "downloads/hotfix/${hotfix_file_name}" "${LIFERAY_DOCKER_TEST_HOTFIX_URL}"
 
 		cp "downloads/hotfix/${hotfix_file_name}" "${TEST_DIR}/patching"
+
+		if [ ! -e "downloads/patching-tool/patching-tool-${LIFERAY_DOCKER_TEST_PATCHING_TOOL_VERSION}.zip" ]
```

Comment:
download checks the existence of the file, I don't think the if is necessary here.

Reply:
Fixed